diff -ur check_radius_ih-1.1/check_radius_ih.c check_radius_ih-1.1.hidden_passwords/check_radius_ih.c
--- check_radius_ih-1.1/check_radius_ih.c	2014-05-24 20:57:32.693522640 +0100
+++ check_radius_ih-1.1.hidden_passwords/check_radius_ih.c	2014-05-24 20:56:42.606152468 +0100
@@ -193,10 +193,12 @@
 			case 'p':		/* set cleartext password			*/
 				escape_semicolons(optarg);
 				strncpy((char *)global.password.pw_clear,optarg,LEGAL_SIZE(global.password.pw_clear));
+				while (*optarg) *optarg++='x';
 				break;
 			case 's':		/* set shared secret				*/
 				escape_semicolons(optarg);
 				strncpy(global.sharedsecret,optarg,LEGAL_SIZE(global.sharedsecret));
+				while (*optarg) *optarg++='x';
 				break;
 			case 'r':		/* set radius server				*/
 				strncpy(global.radiusserver,optarg,LEGAL_SIZE(global.radiusserver));
