diff -ur nagios-plugins-1.4.17-dev/plugins/check_http.c nagios-plugins-1.4.17-dev.with_passwords/plugins/check_http.c
--- nagios-plugins-1.4.17-dev/plugins/check_http.c	2014-05-25 14:14:29.179355489 +0100
+++ nagios-plugins-1.4.17-dev.with_passwords/plugins/check_http.c	2014-05-25 14:08:14.192081056 +0100
@@ -369,10 +369,12 @@
     case 'a': /* authorization info */
       strncpy (user_auth, optarg, MAX_INPUT_BUFFER - 1);
       user_auth[MAX_INPUT_BUFFER - 1] = 0;
+      while (*optarg) *optarg++='x';
       break;
     case 'b': /* proxy-authorization info */
       strncpy (proxy_auth, optarg, MAX_INPUT_BUFFER - 1);
       proxy_auth[MAX_INPUT_BUFFER - 1] = 0;
+      while (*optarg) *optarg++='x';
       break;
     case 'P': /* HTTP POST data in URL encoded format; ignored if settings already */
       if (! http_post_data)
diff -ur nagios-plugins-1.4.17-dev/plugins/check_ldap.c nagios-plugins-1.4.17-dev.with_passwords/plugins/check_ldap.c
--- nagios-plugins-1.4.17-dev/plugins/check_ldap.c	2012-07-31 14:30:56.000000000 +0100
+++ nagios-plugins-1.4.17-dev.with_passwords/plugins/check_ldap.c	2014-05-25 14:10:42.074210311 +0100
@@ -310,7 +310,8 @@
 			ld_binddn = optarg;
 			break;
 		case 'P':
-			ld_passwd = optarg;
+			ld_passwd = strdup(optarg);
+			while (*optarg) *optarg++='x';
 			break;
 		case 'w':
 			warn_time = strtod (optarg, NULL);
diff -ur nagios-plugins-1.4.17-dev/plugins/check_nt.c nagios-plugins-1.4.17-dev.with_passwords/plugins/check_nt.c
--- nagios-plugins-1.4.17-dev/plugins/check_nt.c	2012-07-31 14:30:56.000000000 +0100
+++ nagios-plugins-1.4.17-dev.with_passwords/plugins/check_nt.c	2014-05-25 14:11:17.881760880 +0100
@@ -530,7 +530,8 @@
 				server_address = optarg;
 				break;
 			case 's': /* password */
-				req_password = optarg;
+				req_password = strdup(optarg);
+				while (*optarg) *optarg++='x';
 				break;
 			case 'p': /* port */
 				if (is_intnonneg(optarg))
diff -ur nagios-plugins-1.4.17-dev/plugins/check_pgsql.c nagios-plugins-1.4.17-dev.with_passwords/plugins/check_pgsql.c
--- nagios-plugins-1.4.17-dev/plugins/check_pgsql.c	2012-07-31 14:30:56.000000000 +0100
+++ nagios-plugins-1.4.17-dev.with_passwords/plugins/check_pgsql.c	2014-05-25 14:09:36.103039776 +0100
@@ -289,7 +289,8 @@
 			break;
 		case 'p':     /* authentication password */
 		case 'a':
-			pgpasswd = optarg;
+			pgpasswd = strdup(optarg);
+			while (*optarg) *optarg++='x';
 			break;
 		case 'v':
 			verbose++;
diff -ur nagios-plugins-1.4.17-dev/plugins/check_smtp.c nagios-plugins-1.4.17-dev.with_passwords/plugins/check_smtp.c
--- nagios-plugins-1.4.17-dev/plugins/check_smtp.c	2012-07-31 14:30:56.000000000 +0100
+++ nagios-plugins-1.4.17-dev.with_passwords/plugins/check_smtp.c	2014-05-25 14:13:07.724379554 +0100
@@ -522,7 +522,8 @@
 			authuser = optarg;
 			break;
 		case 'P':
-			authpass = optarg;
+			authpass = strdup(optarg);
+			while (*optarg) *optarg++='x';
 			break;
 		case 'e':									/* server expect string on 220  */
 			server_expect = optarg;
diff -ur nagios-plugins-1.4.17-dev/plugins/check_snmp.c nagios-plugins-1.4.17-dev.with_passwords/plugins/check_snmp.c
--- nagios-plugins-1.4.17-dev/plugins/check_snmp.c	2014-05-25 14:14:29.283354182 +0100
+++ nagios-plugins-1.4.17-dev.with_passwords/plugins/check_snmp.c	2014-05-24 20:39:59.222766560 +0100
@@ -648,7 +648,8 @@
 
 	/* Connection info */
 		case 'C':									/* group or community */
-			community = optarg;
+			community = strdup(optarg);
+			while (*optarg) *optarg++='x';
 			break;
 		case 'H':									/* Host or server */
 			server_address = optarg;
@@ -678,10 +679,12 @@
 			privproto = optarg;
 			break;
 		case 'A':	/* auth passwd */
-			authpasswd = optarg;
+			authpasswd = strdup(optarg);
+			while (*optarg) *optarg++='x';
 			break;
 		case 'X':	/* priv passwd */
-			privpasswd = optarg;
+			privpasswd = strdup(optarg);
+			while (*optarg) *optarg++='x';
 			break;
 		case 't':	/* timeout period */
 			if (!is_integer (optarg))
